pyodide_build/pywasmcross.py

```diff
@@ -578,7 +582,11 @@ def handle_command_generate_args(  # noqa: C901
     # Better to fail at compile or link time.
     if is_link_cmd(line):
         new_args.append("-Wl,--fatal-warnings")
-        new_args.extend(build_args.ldflags.split())
+        for arg in build_args.ldflags.split():
+            if arg.startswith("-l"):
+                arg = replay_genargs_handle_dashl(arg, used_libs, build_args.abi)
```

Comment:
Not a strong opinion, but as `build_args.ldflags` is passed by the user, I think the user is responsible for setting the proper flags.

Reply:
I have mixed feelings. I guess we can't fix static/dynamic libs not built with the python build system anyways so it's not that valuable. But it's pretty weird and annoying that linking `-lpng` successfully links but will break with opaque errors at runtime and the fix is to link with `-lpng-wasm-sjlj`. Maybe we can add some logic to audit wheel or somewhere appropriate that detects files linked against wrong libraries and explains as best we can what went wrong. Or people can just ask. 

So yeah probably it makes sense to remove this.

Reply:
>  Maybe we can add some logic to audit wheel or somewhere appropriate that detects files linked against wrong libraries and explains as best we can what went wrong.

Yeah, I think the problem is that we are not building libraries like `libpng` or `libfreetype` ourselves and use flags like `-s USE_LIBPNG=1` instead. So, there exist multiple variants of those libraries, and it isn't easy to know what to use without looking into the emscripten code.

Ideally, I think it is better to drop all emscripten ports and build libraries ourselves. I think this would fix the issue with the Emscripten version and library version being fixed, and it would also fix the variant issues like the wasm exception.

Reply:
BTW, I noticed that the library suffices are changed in the recent version of Emscripten: https://github.com/emscripten-core/emscripten/blob/242af002f8dee0366e86002fec187d447dcd26e5/tools/ports/libpng.py#L12-L16